src/coreclr/jit/lowerxarch.cpp

```diff
@@ -9369,113 +9369,84 @@ void Lowering::TryFoldCnsVecForEmbeddedBroadcast(GenTreeHWIntrinsic* parentNode,
+    {
+        broadcastName = NI_AVX2_BroadcastScalarToVector256;
+    }
+    else if (simdType == TYP_SIMD64)
     {
-        isCreatedFromScalar = false;
+        broadcastName = NI_AVX512_BroadcastScalarToVector512;
     }
     else
     {
-        isCreatedFromScalar = childNode->IsBroadcast(simdBaseType);
+        assert(simdType == TYP_SIMD16);
     }
 
-    if (isCreatedFromScalar)
+    switch (simdBaseType)
     {
-        NamedIntrinsic broadcastName = NI_AVX2_BroadcastScalarToVector128;
-        if (simdType == TYP_SIMD32)
+        case TYP_FLOAT:
         {
-            broadcastName = NI_AVX2_BroadcastScalarToVector256;
+            float scalar = childNode->gtSimdVal.f32[0];
+            constScalar  = comp->gtNewDconNodeF(scalar);
+            break;
         }
-        else if (simdType == TYP_SIMD64)
+        case TYP_DOUBLE:
         {
-            broadcastName = NI_AVX512_BroadcastScalarToVector512;
+            double scalar = childNode->gtSimdVal.f64[0];
+            constScalar   = comp->gtNewDconNodeD(scalar);
+            break;
         }
-        else
+        case TYP_INT:
+        case TYP_UINT:
         {
-            assert(simdType == TYP_SIMD16);
+            int32_t scalar = childNode->gtSimdVal.i32[0];
+            constScalar    = comp->gtNewIconNode(scalar);
+            break;
         }
-
-        GenTree* constScalar = nullptr;
-        switch (simdBaseType)
+        case TYP_LONG:
+        case TYP_ULONG:
         {
-            case TYP_FLOAT:
-            {
-                float scalar = childNode->gtSimdVal.f32[0];
-                constScalar  = comp->gtNewDconNodeF(scalar);
-                break;
-            }
-            case TYP_DOUBLE:
-            {
-                double scalar = childNode->gtSimdVal.f64[0];
-                constScalar   = comp->gtNewDconNodeD(scalar);
-                break;
-            }
-            case TYP_INT:
-            {
-                int32_t scalar = childNode->gtSimdVal.i32[0];
-                constScalar    = comp->gtNewIconNode(scalar, simdBaseType);
-                break;
-            }
-            case TYP_UINT:
-            {
-                uint32_t scalar = childNode->gtSimdVal.u32[0];
-                constScalar     = comp->gtNewIconNode(scalar, TYP_INT);
-                break;
-            }
-            case TYP_LONG:
-            case TYP_ULONG:
-            {
-                int64_t scalar = childNode->gtSimdVal.i64[0];
-                constScalar    = comp->gtNewLconNode(scalar);
-                break;
-            }
-            default:
-                unreached();
+            int64_t scalar = childNode->gtSimdVal.i64[0];
+            constScalar    = comp->gtNewLconNode(scalar);
+            break;
         }
+        default:
+            unreached();
+    }
 
-        GenTreeHWIntrinsic* createScalar =
-            comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, constScalar, NI_Vector128_CreateScalarUnsafe, simdBaseJitType,
```

Comment:
This cleanup isn't strictly related to the Tier0 enablement, but I noticed this was doing more work than necessary.  It currently creates a `Broadcast <- CreateScalarUnsafe` sequence, which it then lowers.  Lowering removes the `CreateScalarUnsafe` for integral types only, which then requires different logic for marking the child(ren) contained, depending on baseType.

This skips the `CreateScalarUnsafe` since it's never needed.  It also simplifies the operand replacement since we know for sure the CnsVec is used by the parent node.